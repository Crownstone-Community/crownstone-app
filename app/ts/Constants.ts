import { Languages } from "./Languages";

export const DAY_INDICES_SUNDAY_START = [
  "Sun",
  "Mon",
  "Tue",
  "Wed",
  "Thu",
  "Fri",
  "Sat",
]
export const DAY_INDICES_MONDAY_START = [
  "Mon",
  "Tue",
  "Wed",
  "Thu",
  "Fri",
  "Sat",
  "Sun",
]
export const MONTH_INDICES = [
  "Jan",
  "Feb",
  "Mar",
  "Apr",
  "May",
  "Jun",
  "Jul",
  "Aug",
  "Sep",
  "Oct",
  "Nov",
  "Dec",
]

export let WEEK_DAY_INDICES = ['Mon','Tue','Wed','Thu','Fri']; // these are keys
export let WEEKEND_DAY_INDICES = ['Sat','Sun']; // these are keys


export let DAY_LABEL_MAP = function(key) {
  switch (key) {
    case "Mon":
      return Languages.get("__UNIVERSAL", 'DAY_Monday')();
    case "Tue":
      return Languages.get("__UNIVERSAL", 'DAY_Tuesday')();
    case "Wed":
      return Languages.get("__UNIVERSAL", 'DAY_Wednesday')();
    case "Thu":
      return Languages.get("__UNIVERSAL", 'DAY_Thursday')();
    case "Fri":
      return Languages.get("__UNIVERSAL", 'DAY_Friday')();
    case "Sat":
      return Languages.get("__UNIVERSAL", 'DAY_Saturday')();
    case "Sun":
      return Languages.get("__UNIVERSAL", 'DAY_Sunday')();
  }
};

export let DAYS_LABEL_MAP = function(key) {
  switch (key) {
    case "Mon":
      return Languages.get("__UNIVERSAL", 'DAY_Mondays')();
    case "Tue":
      return Languages.get("__UNIVERSAL", 'DAY_Tuesdays')();
    case "Wed":
      return Languages.get("__UNIVERSAL", 'DAY_Wednesdays')();
    case "Thu":
      return Languages.get("__UNIVERSAL", 'DAY_Thursdays')();
    case "Fri":
      return Languages.get("__UNIVERSAL", 'DAY_Fridays')();
    case "Sat":
      return Languages.get("__UNIVERSAL", 'DAY_Saturdays')();
    case "Sun":
      return Languages.get("__UNIVERSAL", 'DAY_Sundays')();
  }
};

export let DAY_SHORT_LABEL_MAP = function(key) {
  switch (key) {
    case "Mon":
      return Languages.get("__UNIVERSAL", 'DAY_Mon')();
    case "Tue":
      return Languages.get("__UNIVERSAL", 'DAY_Tue')();
    case "Wed":
      return Languages.get("__UNIVERSAL", 'DAY_Wed')();
    case "Thu":
      return Languages.get("__UNIVERSAL", 'DAY_Thu')();
    case "Fri":
      return Languages.get("__UNIVERSAL", 'DAY_Fri')();
    case "Sat":
      return Languages.get("__UNIVERSAL", 'DAY_Sat')();
    case "Sun":
      return Languages.get("__UNIVERSAL", 'DAY_Sun')();
  }
};



export let MONTH_LABEL_MAP = function(key) {
  switch (key) {
    case "Jan":
        return Languages.get("__UNIVERSAL", 'MONTH_January')();
    case "Feb":
      return Languages.get("__UNIVERSAL", 'MONTH_February')();
    case "Mar":
      return Languages.get("__UNIVERSAL", 'MONTH_March')();
    case "Apr":
      return Languages.get("__UNIVERSAL", 'MONTH_April')();
    case "May":
      return Languages.get("__UNIVERSAL", 'MONTH_May_full')();
    case "Jun":
      return Languages.get("__UNIVERSAL", 'MONTH_June')();
    case "Jul":
      return Languages.get("__UNIVERSAL", 'MONTH_July')();
    case "Aug":
      return Languages.get("__UNIVERSAL", 'MONTH_August')();
    case "Sep":
      return Languages.get("__UNIVERSAL", 'MONTH_September')();
    case "Oct":
      return Languages.get("__UNIVERSAL", 'MONTH_October')();
    case "Nov":
      return Languages.get("__UNIVERSAL", 'MONTH_November')();
    case "Dec":
      return Languages.get("__UNIVERSAL", 'MONTH_December')();
  }
};



export let MONTH_SHORT_LABEL_MAP = function(key) {
  switch (key) {
    case "Jan":
      return Languages.get("__UNIVERSAL", 'MONTH_Jan')();
    case "Feb":
      return Languages.get("__UNIVERSAL", 'MONTH_Feb')();
    case "Mar":
      return Languages.get("__UNIVERSAL", 'MONTH_Mar')();
    case "Apr":
      return Languages.get("__UNIVERSAL", 'MONTH_Apr')();
    case "May":
      return Languages.get("__UNIVERSAL", 'MONTH_May')();
    case "Jun":
      return Languages.get("__UNIVERSAL", 'MONTH_Jun')();
    case "Jul":
      return Languages.get("__UNIVERSAL", 'MONTH_Jul')();
    case "Aug":
      return Languages.get("__UNIVERSAL", 'MONTH_Aug')();
    case "Sep":
      return Languages.get("__UNIVERSAL", 'MONTH_Sep')();
    case "Oct":
      return Languages.get("__UNIVERSAL", 'MONTH_Oct')();
    case "Nov":
      return Languages.get("__UNIVERSAL", 'MONTH_Nov')();
    case "Dec":
      return Languages.get("__UNIVERSAL", 'MONTH_Dec')();
  }
};

