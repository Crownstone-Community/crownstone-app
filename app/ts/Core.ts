import { eventBus } from "./util/EventBus";
import { NativeBus } from "./native/libInterface/NativeBus";


export const core : core = {
  eventBus: eventBus,
  nativeBus: NativeBus,
  permissionState: {
    location:  'unknown',
    bluetooth: 'unknown',
    bluetoothType: 'SCANNER',
  },
  bleState: {
    bleAvailable: true,
    bleBroadcastAvailable: true,
  },
  store: {
    getState: () : ReduxAppState => {

      // @ts-ignore
      return {}
    },
    dispatch: () => {

    },
    batchDispatch: () => {

    }
  },
};

