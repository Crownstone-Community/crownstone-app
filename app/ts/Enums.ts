export let STONE_TYPES = {
  plug:          "PLUG",
  builtin:       "BUILTIN",
  builtinOne:    "BUILTIN_ONE",
  guidestone:    "GUIDESTONE",
  crownstoneUSB: "CROWNSTONE_USB",
  hub:           "CROWNSTONE_HUB",
};

export const BCH_ERROR_CODES = {
  REMOVED_BECAUSE_IS_DUPLICATE: "REMOVED_BECAUSE_IS_DUPLICATE",
  NO_STONES_FOUND: "NO_STONES_FOUND",
  STONE_IS_LOCKED: "STONE_IS_LOCKED",
  TASK_HAS_BEEN_SUPERSEDED: "TASK_HAS_BEEN_SUPERSEDED",
};


// AiCore types
export const SMART_BEHAVIOUR_TYPES = {
  PRESENCE:      "PRESENCE",
  WAKE_UP_LIGHT: "WAKE_UP_LIGHT",
  SMART_TIMER:   "SMART_TIMER",
  TWILIGHT_MODE: "TWILIGHT_MODE",
  CHILD_SAFETY:  "CHILD_SAFETY",
  CUSTOM:        "CUSTOM",
};

export const AICORE_TIME_TYPES = {
  ALL_DAY:"ALL_DAY",
  RANGE:  "RANGE",
};

export const AICORE_TIME_DETAIL_TYPES = {
  SUNSET:  "SUNSET",
  SUNRISE: "SUNRISE",
  CLOCK:   "CLOCK"
};

export const AICORE_PRESENCE_TYPES = {
  SOMEBODY:       "SOMEBODY",
  NOBODY:         "NOBODY",
  IGNORE:         "IGNORE",
  SPECIFIC_USERS: "SPECIFIC_USERS"
};

export const AICORE_LOCATIONS_TYPES = {
  SPHERE:   "SPHERE",
  LOCATION: "LOCATION",
};

export const SELECTABLE_TYPE = {
  ACTION:      "ACTION",
  PRESENCE:    "PRESENCE",
  LOCATION:    "LOCATION",
  TIME:        "TIME",
  OPTION:      "OPTION",
};

export const KEY_TYPES = {
  ADMIN_KEY:            "ADMIN_KEY",
  MEMBER_KEY:           "MEMBER_KEY",
  BASIC_KEY:            "BASIC_KEY",
  LOCALIZATION_KEY:     "LOCALIZATION_KEY",
  SERVICE_DATA_KEY:     "SERVICE_DATA_KEY",
  MESH_NETWORK_KEY:     "MESH_NETWORK_KEY",
  MESH_APPLICATION_KEY: "MESH_APPLICATION_KEY",
  MESH_DEVICE_KEY:      "MESH_DEVICE_KEY",
  SPHERE_AUTHORIZATION_TOKEN: "SPHERE_AUTHORIZATION_TOKEN",
};

export const BEHAVIOUR_TYPES = {
  twilight: "TWILIGHT",
  behaviour:"BEHAVIOUR",
}

export const HubDataType : HubDataType = {
  SETUP:         0,
  COMMAND:       1,
  FACTORY_RESET: 2,
  REQUEST_DATA:  10,
}

export const HubReplyError : HubReplyError = {
  NOT_IN_SETUP_MODE: 0,
  IN_SETUP_MODE:     1,
  INVALID_TOKEN:     2,
  UNKNOWN:           60000,
}

export const HubReplyCode : HubReplyCode = {
  SUCCESS:    0,
  DATA_REPLY: 10,
  ERROR:      4000
}

export const HubRequestDataType : HubRequestDataType = {
  CLOUD_ID: 0
}


export const CONDITION_MAP = {
  SWITCH_STATE: 'switchState',
};