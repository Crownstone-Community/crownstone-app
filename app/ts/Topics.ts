export const NATIVE_BUS_TOPICS : NativeBusTopics = {
  setupAdvertisement:   "verifiedSetupAdvertisementData",   // data type = crownstoneAdvertisement
  dfuAdvertisement:     "verifiedDFUAdvertisementData",     // data type = crownstoneBaseAdvertisement
  advertisement:        "verifiedAdvertisementData",        // data type = crownstoneAdvertisement // Any verfied advertisement, only normal operation mode.
  crownstoneAdvertisementReceived: "crownstoneAdvertisementReceived",   // data type = crownstoneAdvertisementSummary. // Any advertisement, verified and unverified from crownstones.
  unverifiedAdvertisementData:     "unverifiedAdvertisementData",       // data type = crownstoneAdvertisement // Any unverified advertisement from crownstones.
  setupProgress:        "setupProgress",                    // data type = number ([1 .. 13], 0 for error) // Is 0 required? Or is rejecting the promise enough?
  dfuProgress:          "dfuProgress",                      // data type = {part: number, totalParts: number, progress: number, currentSpeedBytesPerSecond: number, avgSpeedBytesPerSecond: number}
  bleStatus:            "bleStatus",                        // data type = string ("unauthorized", "poweredOff", "poweredOn", "unknown")
  bleBroadcastStatus:   "bleBroadcastStatus",               // data type = string ( "notDetermined" | "restricted" | "denied" | "authorized")
  locationStatus:       "locationStatus",                   // data type = string ("unknown", "off", "foreground", "on", "noPermission")

  nearest:              "nearestCrownstone",                // data type = nearestStone // Any stone, validated or not, any operation mode.
  nearestSetup:         "nearestSetupCrownstone",           // data type = nearestStone

  tick:                 "tick",                             // data type = none. Should be sent about every second.
  iBeaconAdvertisement: "iBeaconAdvertisement",             // data type = ibeaconPackage[]
  enterSphere:          "enterSphere",                      // data type = string (sphereId)
  exitSphere:           "exitSphere",                       // data type = string (sphereId)
  currentRoom:          "currentLocation",                  // Sent every time the location is calculated. data type = {region: sphereId, location: locationId}
  currentLocationKNN:   "currentLocationKNN",               // Sent every time the location is calculated. data type = {region: sphereId, location: locationId}

  libAlert:             "libAlert",                         // data type = {header: string, body: string, buttonText: string }
  libPopup:             "libPopup",                         // data type = {header: string, body: string, buttonText: string, type: <not used yet> }

  classifierProbabilities: "classifierProbabilities",       // data type = {locationId1: {sampleSize: number, probability: number }, locationId2: {sampleSize: number, probability: number }, ...}
  classifierResult:        "classifierResult",              // data type = {highestPredictionLabel: string, highestPrediction: number } // highestPredictionLabel == locationId with highest probability and highestPrediction is that probability

  callbackUrlInvoked:      "callbackUrlInvoked",            // data type = string (url)
  localizationPausedState: "localizationPausedState",       // data type = string (url)

  connectedToPeripheral:       "connectedToPeripheral",       // date type = string (handle). Sent immediately after connect, before service discovery.
  disconnectedFromPeripheral:  "disconnectedFromPeripheral",  // date type = string (handle). Sent after disconnect cleanup. Can be called multiple times.
}

export const TOPICS = {
  datapointCollected: 'datapointCollected',
}
