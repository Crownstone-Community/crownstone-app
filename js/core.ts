import { eventBus } from "./util/EventBus";
import { NativeBus } from "./native/libInterface/NativeBus";


export const core : core = {
  eventBus: eventBus,
  nativeBus: NativeBus,
  bleState: {
    bleAvailable: true,
    bleBroadcastAvailable: true,
  },
  store: { getState: () => { return {}; } },
};

