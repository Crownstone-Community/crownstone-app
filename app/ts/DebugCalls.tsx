import {OnScreenNotifications} from "./notifications/OnScreenNotifications";
import {core} from "./Core";
import {colors, screenHeight, styles} from "./views/styles";
import * as React from "react";
import {NavigationUtil} from "./util/navigation/NavigationUtil";
import {OverlayUtil} from "./views/overlays/OverlayUtil";

export function DebugNotifications() {
  let activeSphereId = core.store.getState().app.activeSphere;

  // OnScreenNotifications.setNotification({
  //   source: "InviteCenterClassX",
  //   id: "invitationToSphere" + activeSphereId,
  //   label: 'invited',
  //   icon: "ios-mail",
  //   callback: () => {
  //     core.eventBus.emit("showCustomOverlay", {
  //       backgroundColor: colors.green.rgba(0.5),
  //     });
  //   }
  // });
  //
  // OnScreenNotifications.setNotification({
  //   source: "BleStateOverlayX",
  //   id: "bluetoothState",
  //   label: "Bluetooth disabled",
  //   icon: "ios-bluetooth",
  //   backgroundColor: colors.csOrange.rgba(0.5),
  //   callback: () => {
  //     NavigationUtil.showOverlay('BleStateOverlay', { notificationType: this.state.notificationType, type: this.state.type });
  //   }
  // })
}

export function DebugOverlays() {
  // let activeSphereId = core.store.getState().app.activeSphere;
  // OverlayUtil.callRoomSelectionOverlay(activeSphereId, (roomId) => {
  //
  // })
}