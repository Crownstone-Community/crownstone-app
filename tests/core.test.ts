import { resetMocks } from "./__testUtil/mocks/suite.mock";
import { addSphere, addStone } from "./__testUtil/helpers/data.helper";
import { core } from "../app/ts/core";

beforeEach(async () => {
  resetMocks()
})
beforeAll(async () => {})
afterEach(async () => { })
afterAll(async () => {})

let handle = 'TestHandle';
test("Using core in tests", async () => {
  addSphere();
  addStone('handle1')
  addStone('handle2')
  addStone('handle3')
  let state = core.store.getState();
  expect(state).toMatchSnapshot()
})