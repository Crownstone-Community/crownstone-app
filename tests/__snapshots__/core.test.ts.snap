// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using core in tests 1`] = `
Object {
  "app": Object {
    "activeSphere": null,
    "hasSeenDeviceSettings": false,
    "hasSeenSwitchView": false,
    "hasZoomedOutForSphereOverview": false,
    "indoorLocalizationEnabled": true,
    "migratedDataToVersion": null,
    "notificationToken": null,
    "tapToToggleEnabled": false,
    "updatedAt": 1,
  },
  "devApp": Object {
    "fastPhone": false,
    "sphereUsedForSetup": "DEV_APP_SPHERE_ID",
    "storeCrownstonesInCloud": false,
  },
  "development": Object {
    "debugRTCtoken": null,
    "devAppVisible": false,
    "firmwareEarlyAccessLevel": 0,
    "log_advertisements": 50,
    "log_ble": 50,
    "log_cloud": 30,
    "log_constellation": 30,
    "log_dfu": 30,
    "log_events": 50,
    "log_info": 30,
    "log_native": 50,
    "log_nav": 30,
    "log_notifications": 30,
    "log_scheduler": 50,
    "log_store": 30,
    "logging_enabled": false,
    "nativeExtendedLogging": false,
    "preview": false,
    "show_rssi_values_in_mesh": false,
    "show_sync_button_in_behaviour": false,
    "use_time_based_nonce": false,
  },
  "devices": Object {},
  "events": Object {
    "remove": Object {
      "behaviours": Object {},
      "devices": Object {},
      "installations": Object {},
      "locations": Object {},
      "messages": Object {},
      "scenes": Object {},
      "stones": Object {},
      "user": Object {},
    },
    "special": Object {
      "behaviours": Object {},
      "devices": Object {},
      "installations": Object {},
      "locations": Object {},
      "messages": Object {},
      "scenes": Object {},
      "stones": Object {},
      "user": Object {},
    },
  },
  "installations": Object {},
  "spheres": Object {
    "sphere_uuid2002-2003": Object {
      "config": Object {
        "aiName": null,
        "cloudId": null,
        "iBeaconUUID": undefined,
        "latitude": null,
        "longitude": null,
        "name": "testSphere",
        "uid": null,
        "updatedAt": 1000000,
      },
      "hubs": Object {},
      "id": "sphere_uuid2002-2003",
      "keys": Object {},
      "locations": Object {},
      "messages": Object {},
      "scenes": Object {},
      "sortedLists": Object {},
      "state": Object {
        "lastPresentTime": 0,
        "newMessageFound": false,
        "present": false,
        "reachable": false,
        "smartHomeEnabled": true,
      },
      "stones": Object {
        "stone_uuid2004-2005": Object {
          "abilities": Object {
            "dimming": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "softOnSpeed": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "softOnSpeed",
                  "updatedAt": 1000000,
                  "value": 8,
                  "valueTarget": 8,
                },
              },
              "syncedToCrownstone": true,
              "type": "dimming",
              "updatedAt": 1000000,
            },
            "switchcraft": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {},
              "syncedToCrownstone": true,
              "type": "switchcraft",
              "updatedAt": 1000000,
            },
            "tapToToggle": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "rssiOffset": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "rssiOffset",
                  "updatedAt": 1000000,
                  "value": 0,
                  "valueTarget": 0,
                },
              },
              "syncedToCrownstone": true,
              "type": "tapToToggle",
              "updatedAt": 1000000,
            },
          },
          "behaviours": Object {},
          "config": Object {
            "bootloaderVersion": null,
            "cloudId": null,
            "description": "",
            "dfuResetRequired": false,
            "firmwareVersion": "5.4.0",
            "handle": "handle_uuid2006",
            "hardwareVersion": null,
            "hidden": false,
            "iBeaconMajor": undefined,
            "iBeaconMinor": undefined,
            "icon": "c2-pluginFilled",
            "locationId": null,
            "locked": false,
            "macAddress": undefined,
            "name": "stone_uuid2007",
            "type": "PLUG",
            "uicr": null,
            "uid": 1,
            "updatedAt": 1000000,
          },
          "errors": Object {
            "dimmerOffFailure": false,
            "dimmerOnFailure": false,
            "hasError": false,
            "overCurrent": false,
            "overCurrentDimmer": false,
            "temperatureChip": false,
            "temperatureDimmer": false,
          },
          "id": "stone_uuid2004-2005",
          "keys": Object {},
          "lastUpdated": Object {
            "stoneTime": 0,
          },
          "reachability": Object {
            "lastSeen": null,
          },
          "state": Object {
            "behaviourOverridden": false,
            "currentUsage": 0,
            "dimmerReady": false,
            "powerFactor": null,
            "previousState": 0,
            "state": 0,
            "timeSet": false,
            "updatedAt": 1,
          },
        },
        "stone_uuid2008-2009": Object {
          "abilities": Object {
            "dimming": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "softOnSpeed": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "softOnSpeed",
                  "updatedAt": 1000000,
                  "value": 8,
                  "valueTarget": 8,
                },
              },
              "syncedToCrownstone": true,
              "type": "dimming",
              "updatedAt": 1000000,
            },
            "switchcraft": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {},
              "syncedToCrownstone": true,
              "type": "switchcraft",
              "updatedAt": 1000000,
            },
            "tapToToggle": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "rssiOffset": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "rssiOffset",
                  "updatedAt": 1000000,
                  "value": 0,
                  "valueTarget": 0,
                },
              },
              "syncedToCrownstone": true,
              "type": "tapToToggle",
              "updatedAt": 1000000,
            },
          },
          "behaviours": Object {},
          "config": Object {
            "bootloaderVersion": null,
            "cloudId": null,
            "description": "",
            "dfuResetRequired": false,
            "firmwareVersion": "5.4.0",
            "handle": "handle_uuid2010",
            "hardwareVersion": null,
            "hidden": false,
            "iBeaconMajor": undefined,
            "iBeaconMinor": undefined,
            "icon": "c2-pluginFilled",
            "locationId": null,
            "locked": false,
            "macAddress": undefined,
            "name": "stone_uuid2011",
            "type": "PLUG",
            "uicr": null,
            "uid": 2,
            "updatedAt": 1000000,
          },
          "errors": Object {
            "dimmerOffFailure": false,
            "dimmerOnFailure": false,
            "hasError": false,
            "overCurrent": false,
            "overCurrentDimmer": false,
            "temperatureChip": false,
            "temperatureDimmer": false,
          },
          "id": "stone_uuid2008-2009",
          "keys": Object {},
          "lastUpdated": Object {
            "stoneTime": 0,
          },
          "reachability": Object {
            "lastSeen": null,
          },
          "state": Object {
            "behaviourOverridden": false,
            "currentUsage": 0,
            "dimmerReady": false,
            "powerFactor": null,
            "previousState": 0,
            "state": 0,
            "timeSet": false,
            "updatedAt": 1,
          },
        },
        "stone_uuid2012-2013": Object {
          "abilities": Object {
            "dimming": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "softOnSpeed": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "softOnSpeed",
                  "updatedAt": 1000000,
                  "value": 8,
                  "valueTarget": 8,
                },
              },
              "syncedToCrownstone": true,
              "type": "dimming",
              "updatedAt": 1000000,
            },
            "switchcraft": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {},
              "syncedToCrownstone": true,
              "type": "switchcraft",
              "updatedAt": 1000000,
            },
            "tapToToggle": Object {
              "cloudId": null,
              "enabled": false,
              "enabledTarget": false,
              "properties": Object {
                "rssiOffset": Object {
                  "cloudId": null,
                  "syncedToCrownstone": true,
                  "type": "rssiOffset",
                  "updatedAt": 1000000,
                  "value": 0,
                  "valueTarget": 0,
                },
              },
              "syncedToCrownstone": true,
              "type": "tapToToggle",
              "updatedAt": 1000000,
            },
          },
          "behaviours": Object {},
          "config": Object {
            "bootloaderVersion": null,
            "cloudId": null,
            "description": "",
            "dfuResetRequired": false,
            "firmwareVersion": "5.4.0",
            "handle": "handle_uuid2014",
            "hardwareVersion": null,
            "hidden": false,
            "iBeaconMajor": undefined,
            "iBeaconMinor": undefined,
            "icon": "c2-pluginFilled",
            "locationId": null,
            "locked": false,
            "macAddress": undefined,
            "name": "stone_uuid2015",
            "type": "PLUG",
            "uicr": null,
            "uid": 3,
            "updatedAt": 1000000,
          },
          "errors": Object {
            "dimmerOffFailure": false,
            "dimmerOnFailure": false,
            "hasError": false,
            "overCurrent": false,
            "overCurrentDimmer": false,
            "temperatureChip": false,
            "temperatureDimmer": false,
          },
          "id": "stone_uuid2012-2013",
          "keys": Object {},
          "lastUpdated": Object {
            "stoneTime": 0,
          },
          "reachability": Object {
            "lastSeen": null,
          },
          "state": Object {
            "behaviourOverridden": false,
            "currentUsage": 0,
            "dimmerReady": false,
            "powerFactor": null,
            "previousState": 0,
            "state": 0,
            "timeSet": false,
            "updatedAt": 1,
          },
        },
      },
      "thirdParty": Object {
        "hue": Object {},
        "toons": Object {},
      },
      "users": Object {},
    },
  },
  "user": Object {
    "accessToken": null,
    "appIdentifier": null,
    "betaAccess": false,
    "bootloaderVersionsAvailable": Object {},
    "developer": false,
    "email": null,
    "firmwareVersionsAvailable": Object {},
    "firstName": null,
    "isNew": true,
    "language": null,
    "lastName": null,
    "passwordHash": null,
    "picture": null,
    "pictureId": null,
    "seenTapToToggle": false,
    "seenTapToToggleDisabledDuringSetup": false,
    "updatedAt": 1,
    "uploadDeviceDetails": true,
    "uploadLocation": true,
    "uploadSwitchState": true,
    "userId": null,
  },
}
`;
